### PR TITLE
add: link Go+ build assets into Go bin path

### PR DESCRIPTION
**Pull request by 92hackers**

### CODE REVIEW

File: cmd/make.go [hunk 1]
```diff
@@ -132,6 +133,62 @@ func detectGopBinPath() string {
...
 
+func detectGoBinPath() string {
+	goBin, ok := os.LookupEnv("GOBIN")
+	if ok {
+		return goBin
+	}
+
+	goPath, ok := os.LookupEnv("GOPATH")
+	if ok {
+		return filepath.Join(goPath, "bin")
```
Comment (xushiwei) [inline cmd/make.go:144]: goPath may be a path list (eg. "a/b:c/d/e").

File: cmd/make.go [hunk 2]
```diff
@@ -132,6 +133,62 @@ func detectGopBinPath() string {
...
+		log.Fatalf("Error: %s is not existed, you should build Go+ before linking.\n", gopBinPath)
+	}
+	if !checkPathExist(goBinPath, true) {
+		if err := os.MkdirAll(goBinPath, 0755); err != nil {
+			fmt.Printf("Error: target directory %s is not existed and we can't create one.\n", goBinPath)
+			log.Fatalln(err)
+		}
+	}
+
+	os.Chdir(gopRoot)
```
Comment (xushiwei) [inline cmd/make.go:166]: don't change directory.

File: cmd/make.go [hunk 3]
```diff
@@ -132,6 +133,62 @@ func detectGopBinPath() string {
...
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	for _, file := range gopBinFiles {
+		if !file.IsDir() {
+			sourceFile := filepath.Join(gopBinPath, file.Name())
+			targetLink := filepath.Join(goBinPath, file.Name())
+			if checkPathExist(targetLink, false) {
+				fmt.Printf("The link file: %s already existed, skip.\n", targetLink)
```
Comment (xushiwei) [inline cmd/make.go:178]: maybe delete is more reasonable.

File: cmd/make.go [hunk 4]
```diff
@@ -132,6 +133,62 @@ func detectGopBinPath() string {
...
+	}
+
+	os.Chdir(gopRoot)
+
+	gopBinFiles, err := os.ReadDir(gopBinPath)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	for _, file := range gopBinFiles {
```
Comment (xushiwei) [inline cmd/make.go:173]: don't need link all apps. just link `gop` and `gopfmt`.

File: cmd/make.go [hunk 5]
```diff
@@ -199,11 +259,33 @@ func runTestcases() {
...
+		return
+	}
+
+	gopBinFiles, err := os.ReadDir(gopBinPath)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	// Clean links
+	for _, file := range gopBinFiles {
```
Comment (xushiwei) [inline cmd/make.go:274]: don't clean links. only delete links when we build success.

File: cmd/make.go [hunk 6]
```diff
@@ -199,11 +259,33 @@ func runTestcases() {
...
+			if checkPathExist(targetLink, false) {
+				if err := os.Remove(targetLink); err != nil {
+					log.Fatalln(err)
+				}
+			}
 		}
 	}
+
+	// Clean build binary files
+	if err := os.RemoveAll(gopBinPath); err != nil {
```
Comment (xushiwei) [inline cmd/make.go:286]: maybe don't remove old results (so that don't break links in goBinPath)?

Comment (92hackers) [reply cmd/make.go:286]: Now, only clean when uninstall.

File: cmd/make.go (hunk 5 above)
Comment (92hackers) [reply cmd/make.go:274]: Now, only clean when uninstall.

Comment (92hackers) [reply cmd/make.go:274]: Now, only clean when uninstall.

File: cmd/make.go (hunk 6 above)
Comment (92hackers) [reply cmd/make.go:286]: Now, only clean when uninstall.

**Pull request merged**